src/humanloop.client.ts

```diff
@@ -0,0 +1,175 @@
+    protected readonly opentelemetryTracer: Tracer;
+
+    constructor(
+        _options: BaseHumanloopClient.Options,
+        OpenAI?: any,
+        Anthropic?: any,
+        CohereAI?: any,
+    ) {
+        super(_options);
+
+        this.OpenAI = OpenAI;
+        this.Anthropic = Anthropic;
+        this.CohereAI = CohereAI;
+
+        this._prompts_overloaded = overloadLog(super.prompts);
+
+        this._flows_overloaded = overloadLog(super.flows);
+
+        this._evaluations = new ExtendedEvaluations(_options, this);
+
+        this.opentelemetryTracerProvider = new NodeTracerProvider({
+            spanProcessors: [
+                new HumanloopSpanProcessor(new HumanloopSpanExporter(this)),
+            ],
+        });
+
+        if (OpenAI) {
+            const instrumentor = new OpenAIInstrumentation({
+                enrichTokens: true,
+            });
+            instrumentor.manuallyInstrument(OpenAI);
+            instrumentor.setTracerProvider(this.opentelemetryTracerProvider);
+            instrumentor.enable();
+        }
+
+        if (Anthropic) {
+            const instrumentor = new AnthropicInstrumentation();
+            instrumentor.manuallyInstrument(Anthropic);
+            instrumentor.setTracerProvider(this.opentelemetryTracerProvider);
+            instrumentor.enable();
+        }
+
+        if (CohereAI) {
+            const instrumentor = new CohereInstrumentation();
+            instrumentor.manuallyInstrument(CohereAI);
+            instrumentor.setTracerProvider(this.opentelemetryTracerProvider);
+            instrumentor.enable();
+        }
+
+        this.opentelemetryTracerProvider.register();
+
+        this.opentelemetryTracer =
+            this.opentelemetryTracerProvider.getTracer("humanloop.sdk");
+    }
+
+    // Check if user has passed the LLM provider instrumentors
+    private _assertProviders() {
+        const noProviderInstrumented = [
+            this.OpenAI,
+            this.Anthropic,
+            this.CohereAI,
+        ].every((p) => !p);
+        if (noProviderInstrumented) {
+            throw new Error(
+                "Using Prompt File utility without passing any provider in the " +
+                    "HumanloopClient constructor. Did you forget to pass them?",
+            );
+        }
+    }
+
+    public prompt<I, M, O>({
+        callable,
+        promptKernel,
+        path,
+    }: {
+        callable: InputsMessagesCallableType<I, M, O>;
+        path: string;
+        promptKernel?: UtilityPromptKernel;
+    }) {
+        this._assertProviders();
+        return promptUtilityFactory(
+            this.opentelemetryTracer,
+            callable,
+            path,
+            promptKernel,
+        );
+    }
+
+    public tool<I, O>({
+        callable,
+        toolKernel,
+        path,
+    }: {
+        callable: ToolCallableType<I, O>;
+        toolKernel: ToolKernelRequest;
+        path?: string;
```

Comment:
why is path required for prompt and flow but not tool?

Reply:
made it mandatory on tool too